--- a/vendor/go.podman.io/image/v5/pkg/sysregistriesv2/system_registries_v2.go
+++ b/vendor/go.podman.io/image/v5/pkg/sysregistriesv2/system_registries_v2.go
@@ -576,6 +576,7 @@
 		Name:                            "registries",
 		Extension:                       "conf",
 		EnvironmentName:                 "CONTAINERS_REGISTRIES_CONF",
+		DoNotLoadDropInFiles:             c.systemRegistriesConfPath != "",
 		RootForImplicitAbsolutePaths:    c.rootForImplicitAbsolutePaths,
 		CustomConfigFilePath:            c.systemRegistriesConfPath,
 		CustomConfigFileDropInDirectory: c.systemRegistriesConfDirPath,
